src/app/components/SubmitButton/SubmitButton.tsx

```diff
@@ -20,9 +28,36 @@ export default function SubmitButton({
   icon,
   children,
 }: SubmitButtonProps): JSX.Element {
+  let iconElement;
+  switch (icon) {
+    case 'arrowRight':
+      iconElement = <ArrowRightIcon />;
+      break;
+    case 'camera':
+      iconElement = <CameraIcon />;
+      break;
+    case 'letter':
+      iconElement = <LetterIcon />;
+      break;
+    case 'safe':
+      iconElement = <SafeIcon />;
+      break;
+    case 'search':
+      iconElement = <SearchIcon />;
+      break;
+    case 'send':
+      iconElement = <SendIcon />;
+      break;
+    case 'signup':
+      iconElement = <SignUpIcon />;
+      break;
+    case 'upload':
+      iconElement = <UploadIcon />;
+      break;
+  }
```

Comment:
nice, but a map would be a bit more beautiful

Reply:
Konni said the same last night and then I hissed at him like a cat with rabies. I will consider rewriting it if I have time :D